engine/src/main/java/org/terasology/engine/bootstrap/EnvironmentSwitchHandler.java

```diff
@@ -123,6 +119,15 @@ public void handleSwitchToGameEnvironment(Context context) {
 
     }
 
+    @SuppressWarnings("unchecked")
+    private <T, U extends CopyStrategy<?>> void registerCopyStrategy(CopyStrategyLibrary copyStrategyLibrary, Class<T> type, Class<U> strategy) {
```

Comment:
Any reason why this needs to be moved to its own function?

Reply:
The call to CopyStrategyLibrary.register had two different wildcards captured by the same type argument. This is not supposed to compile successfully (and IDEA reports this), but does currently for whatever reason. The fix is to capture each wildcard in a separate argument, which I've done with this function.

Reply:
Good catch but change the second type to `U extends CopyStrategy<T>`. This will save you the cast in line 125 and fits the signature of CopyStrategyLibrary.register: `public <T> void register(Class<T> type, CopyStrategy<T> strategy)`. Then this PR is good to go =)

Reply:
It just makes the cast implied in the call, but it cleans up the signature of `registerCopyStrategy` so I went ahead and did it.